README.md

```diff
@@ -81,32 +86,36 @@ segment`/users/${arg("userId", {
 
 When creating a route, path parameters can be passed in the first argument:
 
-```js
+```ts
 routes.users({ userId: "10" });
 ```
 
 There are some predefined convenience parameter types provided:
 
--   `number(name: string, optional?: boolean)` for number strings
--   `uuid(name: string, optional?: boolean)` for UUID strings
+-   `string(name: string, { optional?: boolean })` for plain strings
+-   `number(name: string, { optional?: boolean })` for number strings
+-   `uuid(name: string, { optional?: boolean })` for UUID strings
 
 ### Query string
 
-Query string parameters can be specified by interpolating `query` function inside a segment string. The `query` function
+Query string parameters can be specified by adding `query` property to the route description. The `query` function
 expects an object where keys are names of parameters and values specify whether those params are required in the path.
 
-```js
-segment`/product${query({
-    productId: true,
-    details: false,
-})}`;
+```ts
+{
+    ...segment`/product`,
+    query: {
+        productId: query(false),
+        details: query(true)
```

Comment:
The new `"optional"`/`"required"` params instead of `true`/`false` are already implemented, right?

Reply:
Tests passed but I didn't update the docs :P Nice catch.